Error compiling 4.1.0 source code with mingw-w64: 'D3D11_TEXTURE2D_DESC' was not declared in this scope

##### System information (version)
- OpenCV => 4.1.0
- Operating System / Platform => Windows 10 64bit
- Compiler => mingw-w64 8.1.0-posix-seh-rt_v6-rev0

##### Detailed description

When I compiled the 4.1.0 version with mingw-w64, there was an error 

```
E:\opencv-4.1.0\opencv-4.1.0\modules\core\src\directx.cpp:1035:5: error: 'D3D11_TEXTURE2D_DESC' was not declared in this scope
     D3D11_TEXTURE2D_DESC desc = { 0 };
     ^~~~~~~~~~~~~~~~~~~~
```

I can solve it like this

```
cmake -DWITH_OPENCL_D3D11_NV=OFF
```
But I don't know if this is the right solution, and whether this will affect the compiled dll.

--- a/cmake/OpenCVDetectDirectX.cmake
+++ b/cmake/OpenCVDetectDirectX.cmake
@@ -22,4 +22,9 @@ if(WIN32)
   set(HAVE_D3D11 ON)
   set(HAVE_D3D10 ON)
   set(HAVE_D3D9 ON)
+
+  if(HAVE_OPENCL AND WITH_OPENCL_D3D11_NV AND EXISTS "${OPENCL_INCLUDE_DIR}/CL/cl_d3d11_ext.h")
+    set(HAVE_OPENCL_D3D11_NV ON)
+  endif()
+
 endif()

--- a/cmake/OpenCVDetectOpenCL.cmake
+++ b/cmake/OpenCVDetectOpenCL.cmake
@@ -11,10 +11,6 @@ mark_as_advanced(OPENCL_INCLUDE_DIR OPENCL_LIBRARY)
 
 if(OPENCL_FOUND)
 
-  if(WITH_OPENCL_D3D11_NV AND EXISTS "${OPENCL_INCLUDE_DIR}/CL/cl_d3d11_ext.h")
-    set(HAVE_OPENCL_D3D11_NV ON)
-  endif()
-
   if(OPENCL_LIBRARY)
     set(HAVE_OPENCL_STATIC ON)
     set(OPENCL_LIBRARIES "${OPENCL_LIBRARY}")
@@ -82,4 +78,7 @@ if(OPENCL_FOUND)
       list(APPEND OPENCL_INCLUDE_DIRS "${CLAMDBLAS_INCLUDE_DIR}")
     endif()
   endif()
+
+  # check WITH_OPENCL_D3D11_NV is located in OpenCVDetectDirectX.cmake file
+
 endif()

